skimage/filters/_gabor.py

```diff
@@ -28,21 +206,25 @@ def gabor_kernel(frequency, theta=0, bandwidth=1, sigma_x=None, sigma_y=None,
     ----------
     frequency : float
         Spatial frequency of the harmonic function. Specified in pixels.
-    theta : float, optional
+    theta : float or array of floats, optional
         Orientation in radians. If 0, the harmonic is in the x-direction.
     bandwidth : float, optional
-        The bandwidth captured by the filter. For fixed bandwidth, `sigma_x`
-        and `sigma_y` will decrease with increasing frequency. This value is
-        ignored if `sigma_x` and `sigma_y` are set by the user.
-    sigma_x, sigma_y : float, optional
-        Standard deviation in x- and y-directions. These directions apply to
-        the kernel *before* rotation. If `theta = pi/2`, then the kernel is
-        rotated 90 degrees so that `sigma_x` controls the *vertical* direction.
+        The bandwidth captured by the filter. For fixed bandwidth, `sigma`
+        will decrease with increasing frequency. This value is ignored if
+        `sigma` are set by the user.
+    sigma : float or array of floats, optional
+        Standard deviation. These directions apply to the kernel *before*
+        rotation.
     n_stds : scalar, optional
         The linear size of the kernel is n_stds (3 by default) standard
         deviations
     offset : float, optional
         Phase offset of harmonic function in radians.
+    leading_axis : int, optional
+        The leading axis for rotation. Defaults to 1, the `x` coordinate
+        in most systems.
```

Comment:
Hmm, since this is new functionality, we don't need to worry too much about prior conventions, and I would prefer for the `leading_axis=0` convention to be the default. I think that is the case for e.g. scipy.ndimage rotations. Could you check?

Reply:
@jni This was more put in place for backwards compatibility due to the aforementioned issues with quasipolar decomposition and the ordering of axes.

Come to think of it, with all of the poor design choices I've had to make to preserve backwards compatibility, it might be easier to just make this a completely new function.